src/interactive-window/types.ts

```diff
@@ -76,6 +78,22 @@ export interface IInteractiveWindow extends IInteractiveBase {
     export(cells?: ICell[]): void;
 }
 
+export interface IInteractiveWindowCache {
+    owner: Resource;
+    mode: InteractiveWindowMode;
+    uri: Uri;
+    inputBoxUri: Uri;
```

Comment:
I don't think we should be storing Uris into memento, they don't get serialized and deserialized properly, its stored as plain JSON and this could lead to problems

Reply:
Valid point. Another approach is introducing `UriComponents` like in the core, where `UriComponents` is an interface. But function-wise it works the same as `string`.